camel/tasks/task.py

```diff
@@ -0,0 +1,281 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+
+import re
+from enum import Enum
+from typing import Callable, Dict, List, Optional, Union
+
+from pydantic import BaseModel
+
+from camel.agents import ChatAgent
+from camel.messages import BaseMessage
+from camel.prompts import TextPrompt
+
+from .task_prompt import TASK_DECOMPOSE_PROMPT, TASK_EVOLVE_PROMPT
+
+
+class TaskSate(str, Enum):
+    OPEN = "OPEN"
+    RUNNING = "RUNNING"
+    DONE = "DONE"
+    DELETED = "DELETED"
+
+    @classmethod
+    def states(cls):
+        return [s.value for s in cls]
+
+
+class Task(BaseModel):
+    """
+    Task is specific assignment that can be passed to a agent.
+
+    Attributes:
+        content: string content for task.
+        id: A unique id.
+        state: The state which should be OPEN, RUNNING, DONE or DELETED.
+        parent: The parent task, None for root task.
+        subtasks: The childrent sub-tasks for the task.
+        result: The answer for the task.
```

Comment:
1. how about add expected result as attribute to help agent understand `Task` requirement better?
2. the content may not only include pure text, it also could be path for multi-modal information like image and video, this would also be covered in `Message` design
3. should we also allow user upload their own information like local pdf file? If this task is to handle some provided knowledge base

Reply:
Hi, @Wendong-Fan , here the `TaskManager` is used as utils to mange `Tasks`, It' not conflict with task-related agents. The task-related agents should use `TaskManager` to do some management.

Reply:
1. And for the `expected result`, I think if we have the labeled data that would be useful. If needed I can add it as a optional attribute.
2. the multi-modal information should be added, but how to use if should be updated in the next version.